shoes-core/spec/shoes/builtin_methods_spec.rb

```diff
@@ -106,4 +78,49 @@
       end
     end
   end
+
+  %w(alert confirm).each do |type|
+    describe '##{alert}' do
+     let(:dialog) { double('dialog') }
+     let(:message) { double('message') }
+
+      before do
+        allow(Shoes::Dialog).to receive(:new).and_return(dialog)
+        expect(dialog).to receive(type).with(message).and_return(dialog)
+      end
+
+      it 'creates a new dialog' do
+        expect(app.send(type, message)).to eq(dialog)
+      end
+    end
+  end
+
+  describe '#ask' do
+    let(:dialog) { double('dialog') }
+    let(:message) { double('message') }
+    let(:args) { double('args') }
+
+    before do
+      allow(Shoes::Dialog).to receive(:new).and_return(dialog)
+      expect(dialog).to receive(:ask).with(message, args).and_return(dialog)
+    end
+
+    it 'creates a new dialog' do
+      expect(app.send(:ask, message, args)).to eq(dialog)
+    end
+  end
+
+  describe '#ask_color' do
+    let(:dialog) { double('dialog') }
+    let(:title) { double('title') }
+
+    before do
+      allow(Shoes::Dialog).to receive(:new).and_return(dialog)
```

Comment:
rather then a wrapping context I would like to propose to move this to the top, so the dialog double is only defined once, costwise this should not impact the other test which don't really need the double but will leave a more readable testsuite though (not hiding method describe blocks within the shared context)